plugins/anx-udev/fu-anx-udev-device.c

```diff
@@ -0,0 +1,475 @@
+    guint8 fw_ver[2];
+	guint8 cus_ver[2];
+	guint16 fw_i_ver = 0;
+	guint16 cus_i_ver = 0;
+    g_autofree gchar *version = NULL;
+    /*get OCM version*/
+    if(!fu_anx_udev_device_receive (self, ANX_BB_RQT_READ_FW_VER, 0, 0,
+									&fw_ver[1], 1, error))
+        return FALSE;
+
+    if(!fu_anx_udev_device_receive(self, ANX_BB_RQT_READ_FW_RVER, 0, 0,
+												&fw_ver[0], 1, error))
+        return FALSE;
+	fw_i_ver = (fw_ver[1] << 8) | fw_ver[0];
+	cus_i_ver = (cus_ver[1] << 8) | cus_ver[0];
+    version = g_strdup_printf ("%04x.%04x", cus_i_ver, fw_i_ver);
+    fu_device_set_version (FU_DEVICE (device), version);
+	set_custom_version (cus_i_ver);
+	set_fw_version (fw_i_ver);
+    return TRUE;
+}
+
+static gboolean
+fu_anx_udev_device_find_interface (FuUsbDevice *device,
+                                      GError **error)
+{
+	GUsbDevice *usb_device = fu_usb_device_get_dev (device);
+	FuAnxUdevDevice *self = FU_ANX_UDEV_DEVICE (device);
+	g_autoptr(GPtrArray) intfs = NULL;
+	/* based on usb_updater2's find_interfacei() and find_endpoint() */
+
+	intfs = g_usb_device_get_interfaces (usb_device, error);
+	if (intfs == NULL)
+	{
+		g_debug ("anx_udev:fu_anx_udev_device_find_interface,intfs == NULL");
+		return FALSE;
+	}
+	self->vid = g_usb_device_get_vid (usb_device);
+	self->pid = g_usb_device_get_pid (usb_device);
+	self->rev = g_usb_device_get_release (usb_device);
+	g_debug ("USB: VID:%04X, PID:%04X, REV:%04X", self->vid, self->pid,
+					self->rev);
+	for (guint i = 0; i < intfs->len; i++) {
+		GUsbInterface *intf = g_ptr_array_index (intfs, i);
+		if (g_usb_interface_get_class (intf) == BILLBOARD_CALSS &&
+			g_usb_interface_get_subclass (intf) == BILLBOARD_SUBCLASS &&
+			g_usb_interface_get_protocol (intf) == BILLBOARD_PROTOCOL) {
+			//GUsbEndpoint *ep;
+			g_autoptr(GPtrArray) endpoints = NULL;
+
+			endpoints = g_usb_interface_get_endpoints (intf);
+			if (NULL == endpoints)
+					continue;
+			/*if (endpoints->len == 0)
+			{
+					ep = endpoints;
+			}
+			else
+			{
+					ep = g_ptr_array_index (endpoints, 0);
+			}
+			g_debug ("anx_udev:fu_anx_udev_device_find_interface, NULL != endpoints");
+			self->iface_idx = g_usb_interface_get_number (intf);
+			self->ep_num = g_usb_endpoint_get_address (ep) & 0x7f;
+			self->chunk_len = g_usb_endpoint_get_maximum_packet_size (ep);
+            */
+			return TRUE;
+		}
+	}
+	g_set_error_literal (error,
+						FWUPD_ERROR,
+						FWUPD_ERROR_NOT_FOUND,
+						"no update interface found");
+	return FALSE;
+}
+
+static gboolean
+fu_anx_udev_device_probe (FuUsbDevice *device, GError **error)
+{
+	g_debug ("anx_udev:fu_anx_udev_device_probe");
+	if (!fu_anx_udev_device_find_interface (device, error)) {
+		g_prefix_error (error, "failed to find update interface: ");
+		return FALSE;
+	}
+	set_custom_version (0);
+	set_fw_version (0);
+	/* set name and vendor */
+	fu_device_set_summary (FU_DEVICE (device),
+							"Analogix Phoenix-Lite");

```

Comment:
No need for the Vendor name there, otherwise we'll show "Analogix Analogix Phoenix-Lite"